integrationtest/showcase/library/src/test/java/org/hibernate/search/integrationtest/showcase/library/functional/LessMemoryEntityService.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.showcase.library.functional;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+
+import org.hibernate.search.integrationtest.showcase.library.model.Book;
+import org.hibernate.search.integrationtest.showcase.library.model.BookCopy;
+import org.hibernate.search.integrationtest.showcase.library.model.BookMedium;
+import org.hibernate.search.integrationtest.showcase.library.model.Library;
+import org.hibernate.search.integrationtest.showcase.library.model.LibraryServiceOption;
+import org.hibernate.search.integrationtest.showcase.library.repository.DocumentCopyRepository;
+import org.hibernate.search.integrationtest.showcase.library.repository.DocumentRepository;
+import org.hibernate.search.integrationtest.showcase.library.repository.LibraryRepository;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Similar to {@link org.hibernate.search.integrationtest.showcase.library.service.LibraryService} and
+ * {@link org.hibernate.search.integrationtest.showcase.library.service.DocumentService}.
+ * <p>
+ * But it applies also a <b>flush</b> and a <b>clear</b> on {@link EntityManager} and an <b>evictAll</b> on {@link EntityManagerFactory#getCache()}
+ * before every commit.
+ * <p>
+ * This strategy can be useful for a batch processing to release memory.
+ */
+@Service
+@Transactional
+public class LessMemoryEntityService {
```

Comment:
If we really want a functional IT, and if you really want to put it in the showcase, I think we need to be closer to the use case.

Unless I'm mistaken, the use case was batch inserts. So, something like a `BatchInsertService` or a `DatabaseInitializationService` a `batchInsert(Iterator<BookToInsert>)` method in `DocumentService` (`BookToInsert` being a bean created from an external source) would seem to make more sense.

The alternative, of course, would be to move this test to `org.hibernate.search.integrationtest.mapper.orm.nonregression.automaticindexing`. Then, you could just have all the flush/clear code in a single method, and the entity classes in the same `.java` file. It would probably be easier.

Reply:
I would opt the latter. But would it still a functional?

Reply:
To me it is, since it tests a feature. Thanks.